datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -1124,7 +1124,27 @@ impl LogicalPlanBuilder {
         )?))
     }
 }
-
+pub fn change_redundant_column(fields: Vec<DFField>) -> Vec<DFField> {
+    let mut name_map = HashMap::new();
+    fields
+        .into_iter()
+        .map(|field| {
+            let counter = name_map.entry(field.name().to_string()).or_insert(0);
+            *counter += 1;
+            if *counter > 1 {
+                let new_name = format!("{}:{}", field.name(), *counter - 1);
```

Comment:
This effectively renames some of the columns to something else (like `a:1`) I think

How do we:
1. know that name is unique? (couldn't there be another table in the query that has a column named `a:1`?
2. know we should resolve the namespace conflict with the first column with the name?